commit addf46e590b0bcf8c0fe60409938b64667c98baf
Author: Andrei Sabalenka <mechakotik@gmail.com>
Date:   Wed Aug 21 23:35:47 2024 +0300

    Qbs: Add option to specify custom Python pkg-config name
    
    I am Gentoo maintainer of Tiled. In Gentoo, there may be many Python versions installed at the same time for compatibility reasons, and each version has separate name in pkg-config.
    
    For example, for Python 3.12 it would be named python-3.12-embed, so running pkg-config for python-embed will fail. Adding this option so it's possible to specify the correct name from build script.

https://github.com/mapeditor/tiled/commit/7969644284ae0a1d09b3ab14a7a68f65c79e7976
https://github.com/mapeditor/tiled/pull/4039

diff --git a/src/plugins/python/python.qbs b/src/plugins/python/python.qbs
index f1c959d3..abe9403f 100644
--- a/src/plugins/python/python.qbs
+++ b/src/plugins/python/python.qbs
@@ -22,7 +22,7 @@ TiledPlugin {
 
     Probes.PkgConfigProbe {
         id: pkgConfigPython3
-        name: "python3-embed"
+        name: project.pythonPkgConfigName
         minVersion: "3.8"
     }
 
diff --git a/tiled.qbs b/tiled.qbs
index c8d5018d..fb54d9ba 100644
--- a/tiled.qbs
+++ b/tiled.qbs
@@ -18,6 +18,7 @@ Project {
     property bool sentry: false
     property bool dbus: true
     property string openSslPath: Environment.getEnv("OPENSSL_PATH")
+    property string pythonPkgConfigName: "python3-embed"
 
     references: [
         "dist/archive.qbs",
